materialize-rockset/driver.go

```diff
@@ -210,6 +278,25 @@ func (d *rocksetDriver) Transactions(stream pm.Driver_TransactionsServer) error
 		return err
 	}
 
+	// It's possible that there was a previous materialization with the same name as this one, and that it may have left
+	// a driver checkpoint. This is expected in the case that a user wishes to use a cloud storage integration in
+	// Rockset for bulk loading data for a large backfill as described here: https://go.estuary.dev/rock-bulk
+	// If such a checkpoint is left over, then we'll return it here. The driver checkpoint will then be cleared out
+	// on the next successful commit of a transaction.
+	var flowCheckpoint []byte
+	var s3DriverCheckpoint checkpoint.DriverCheckpoint
+	if len(open.Open.DriverCheckpointJson) > 0 {
```

Comment:
Good question, I had to look back at the code to verify that this does what I thought it did. It effectively just takes the the new value, which is an empty byte slice. It does this because we _don't _ use merge patches for materialization checkpoints. [Here's the code](https://github.com/estuary/protocols/blob/main/flow/checkpoint_extensions.go) that does the reduction of the old and new checkpoints. For materializations, `Rfc7396MergePatch` is always `false`.